app/hooks/useScript.ts

```diff
@@ -0,0 +1,113 @@
+import { useEffect, useState } from 'react';
```

Comment:
Where did we get this hook from?

Reply:
https://usehooks-ts.com/react-hook/use-script slightly modified from here.